### PR TITLE
fix: Make code examples nicer, fix some bugs

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Adds a lot more functionality to our code examples so they are accurate for any use case

Adds support for groups, payloads with local eval, and deep linking documentation

## Changes

Draft still, got to convert payloads as well.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by annikaschmid — APPROVED** [review]
Looks good from what I can see.

File: frontend/src/scenes/feature-flags/FeatureFlagSnippets.tsx [hunk 1]
```diff
@@ -1,130 +1,327 @@
...
+    localEvaluation?: boolean
+    payload?: boolean
+}
+
+export function NodeJSSnippet({
+    flagKey,
+    groupType,
+    multivariant,
+    localEvaluation,
+    payload,
```
Comment (liyiy) [inline frontend/src/scenes/feature-flags/FeatureFlagSnippets.tsx:21]: nit: I think forcing all these options into one code snippet definitely makes it more unreadable and harder to parse

Comment (liyiy) [reply frontend/src/scenes/feature-flags/FeatureFlagSnippets.tsx:21]: so I don't love it but if it works then 👍

**Review by liyiy — APPROVED** [review]
looks good 🟢

File: frontend/src/scenes/feature-flags/FeatureFlagSnippets.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/feature-flags/FeatureFlagSnippets.tsx:21]: Agree it's a trade-off, although I prefer this for maintainability + extending things.

**Pull request merged**